lib/rubocop/cop/factory_bot/association_style.rb

```diff
@@ -218,6 +224,14 @@ def options_for_autocorrect_to_implicit_style(node)
           end
           options
         end
+
+        def trait_within_trait?(node)
+          node.each_ancestor(:block).any? do |ancestor|
+            if ancestor.send_node.method?(:factory)
+              trait_name(ancestor).include?(node.method_name)
+            end
+          end
```

Comment:
Performance wise, shouldn't we compile a list of traits on the very definition of the factory, and then reuse it. Or just go to the top block, as now you're traveling part of the tree every time you are going up (when you want to look also into the siblings)

Reply:
I updated this PR. Thank you.